integrations/terraform/DOCS.md

```diff
@@ -0,0 +1,56 @@
+## Teleport Terraform Provider documentation
+
+
+### What documentation is available
+
+We maintain:
+- [The Terraform Provider reference](https://goteleport.com/docs/reference/terraform) describing every resource and supported field.
+- [The installation guide](https://goteleport.com/docs/management/dynamic-resources/terraform-provider/)
+- A getting started guide describing [how to configure users and roles with Terraform](https://goteleport.com/docs/management/dynamic-resources/user-and-role/)
+
+### Contributing to the documentation
+
+#### Building the reference
+
+The full build process involves:
+- a Makefile target `docs` that re-builds and installs the provider locally
+- a bash script cretaing a temporary directory, exporting the schema from the built provider, invoking `tfplugindocs`
+  and copying the reference into the mains docs directory.
+- [a custom version of `tfplugindocs`](https://github.com/gravitational/terraform-plugin-docs) which renders markdown
+  compatible with our documentation engine.
+
+To re-render the reference, run `make docs`.
```

Comment:
When adding a new resource, table of contents in config.js needs to be updated manually, correct? Might wanna mention it here just in case.